### DISCUSSION TITLE
Confusion with Streaming training and decoding options

### DISCUSSION THREAD

**Discussion by AmirHussein96** [k2-fsa/icefall#744] [Q&A] [answered]
I recently looked into streaming options and got confused specially for the stateless transducer. 
For training I found that there were new options added: 

```
 --dynamic-chunk-training 1 \
  --causal-convolution 1 \
  --short-chunk-size 20 \
  --num-left-chunks 4 

```

Does this mean that if I have an old transducer that was trained before these option that I need to train a new one adding these options for streaming?

My second question is related to decoding:

I I want to test my model in real streaming scenario on my test set that was segmented and transcribed by human do I use the `decode.py` with:

```
--simulate-streaming 1 \
    --causal-convolution 1 \
    --decode-chunk-size 16 \
    --left-context 64 \
```

or should I use the `streaming_decode.py` with :

```
      --left-context 32 \
      --right-context 0 \
      --decode-chunk-size 8 \
      --num-decode-streams 200 \

```

**Comment by pkufool** [comment, marked as answer]
> Does this mean that if I have an old transducer that was trained before these option that I need to train a new one adding these options for streaming?

Yes.

About the decoding, I think you can use either `decode.py` or `streaming_decode.py`, they should produce the same results given the same `decode-chunk-size` and `left-context`.  The difference between `decode.py` and `streaming_decode.py` is that, `decode.py` processes the whole acoustic frames at one time with masking (i.e. same as training), but `streaming_decode.py` processes the acoustice frames chunk by chunk (so it can only see limited context).

**Reply by AmirHussein96** [reply]
Thanks a lot @pkufool for the detailed answer.
